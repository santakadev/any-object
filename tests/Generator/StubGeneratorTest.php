<?php

namespace Santakadev\AnyObject\Tests\Generator;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Santakadev\AnyObject\Generator\StubGenerator;
use Santakadev\AnyObject\Tests\Generator\Generated\Any;
use Santakadev\AnyObject\Tests\TestData\ScalarTypes\IntObject;
use Santakadev\AnyObject\Tests\TestData\ScalarTypes\StringIntObject;
use Santakadev\AnyObject\Tests\TestData\ScalarTypes\StringObject;

class StubGeneratorTest extends TestCase
{
    public function test_generator_string(): void
    {
        $generator = new StubGenerator();
        $text = $generator->generate(StringObject::class);
        Approvals::verifyString($text);
        $test = Any::of(StringObject::class)->with(value: "test");
        $this->assertEquals("test", $test->value);
    }

    public function test_generator_int(): void
    {
        $generator = new StubGenerator();
        $text = $generator->generate(IntObject::class);
        Approvals::verifyString($text);
        $test = Any::of(IntObject::class)->with(1);
        $this->assertEquals(1, $test->value);
    }

    public function test_generator_string_int(): void
    {
        $generator = new StubGenerator();
        $text = $generator->generate(StringIntObject::class);
        Approvals::verifyString($text);
        $test = Any::of(StringIntObject::class)->with("str", 1);
        $this->assertEquals("str", $test->str);
        $this->assertEquals(1, $test->number);
        $this->assertIsString(Any::of(StringIntObject::class)->build()->str);
        $this->assertIsInt(Any::of(StringIntObject::class)->build()->number);
    }
}
