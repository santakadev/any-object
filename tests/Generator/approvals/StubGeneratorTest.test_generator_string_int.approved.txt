<?php

namespace Santakadev\AnyObject\Tests\Generator\Generated;

use Faker\Factory;
use Santakadev\AnyObject\Tests\TestData\ScalarTypes\StringIntObject;
final class StringIntObjectFactory
{
    public function with(string|ValueNotProvided $str = new ValueNotProvided(), int|ValueNotProvided $number = new ValueNotProvided()) : StringIntObject
    {
        if ($str instanceof ValueNotProvided) {
            $faker = Factory::create();
            $str = $faker->text();
        }
        if ($number instanceof ValueNotProvided) {
            $faker = Factory::create();
            $number = $faker->numberBetween(PHP_INT_MIN, PHP_INT_MAX);
        }
        return new StringIntObject($str, $number);
    }
    public function build() : StringIntObject
    {
        return $this->with();
    }
}
